c10/core/WrapDimMinimal.h

```diff
@@ -22,4 +25,18 @@ static inline int64_t maybe_wrap_dim(
   return c10::detail::maybe_wrap_dim_slow(dim, dim_post_expr, wrap_scalar);
 }
 
+static inline SymInt maybe_wrap_dim_symint(
+    SymInt dim,
+    SymInt dim_post_expr,
+    bool wrap_scalar = true) {
+  // Inline the fast paths
+  if (C10_LIKELY(dim_post_expr * -1 <= dim && dim < dim_post_expr)) {
+    // Branch-less version of dim + (dim < 0 ? dim_post_expr : 0)
+    return dim + dim_post_expr * (dim < 0);
```

Comment:
@Chillee can you take a look at these

Reply:
This doesn't matter, since these are on dims (which are fixed anyways).

Reply:
eh.. check the call site of maybe_wrap_dim inside narrow.  and check the call site of narrow from cat.

Reply:
Isn't this a hack lol? This was written for dim wrapping, but the call-site is just using it for shape wrapping.

In general, I'd probably prefer just having a guard here. Technically, that's what this does today (since boolean comparisons immediately guard), but I'd be fine with writing this in the more natural way.

Reply:
i suspect you're right. I probably should have left the dim arg as int, and put a guard in cat.  (is that what you meant?)

Reply:
although tbc, would I still want dim_post_expr to be sym, and keep (ab)using the wrap_dim for wrap_shape, or not?

Reply:
> although tbc, would I still want dim_post_expr to be sym, and keep (ab)using the wrap_dim for wrap_shape, or not?

Yeah I think that's fine. Basically, TL;DR: Rewrite this to more explicitly use control flow.